github/codeowners.py

```diff
@@ -81,73 +111,173 @@ def filter_codeowners_entries(
         yield from line.split()[1:]
 
 
-def _first(
-    iterable: typing.Iterable,
-):
+def resolve_team_members(
+    team_name: str,
+    org: github3.orgs.Organization,
+    absent_ok: bool=False,
+) -> typing.Generator[github3.users.ShortUser, None, None]:
+    '''
+    Returns generator yielding members from team.
+    Absent teams passed silently with `absent_ok`.
+    Subteams are resolved as well.
+    '''
     try:
-        return next(iterable)
-    except StopIteration:
-        return None
+        team = org.team_by_name(team_name)
+    except github3.exceptions.NotFoundError:
+        if absent_ok:
+            return
+        raise
+    # resolves subteams as well
+    yield from team.members()
 
 
-def determine_email_address(
-    github_user_name: str,
-    github_api: GitHub,
-) -> typing.Optional[str]:
-    not_none(github_user_name)
+def mail_address_for_user(
+    user_name: str,
+    gh_api: GitHub | GitHubEnterprise,
+    absent_ok: bool=True,
+) -> str | None:
+    '''
+    Returns email associated with github user for username.
+    `None` returned if user not found and `absent_ok`.
+    Users without associated email return `None`.
```

Comment:
please be accurate w/ comments. it is not true users do not have an "assiciated email [address]". instead, users may choose to not expose a public email address; they always have at least one email-address (it is possible to have multiple email-addresses, with one of them being the "primary" one which is used e.g. for passwd-resets).